components/financial_assistance/app/models/financial_assistance/applicant.rb

```diff
@@ -725,6 +725,7 @@ def applicant_validation_complete?
       if is_applying_coverage
         valid?(:submission) &&
           incomes.all? {|income| income.valid? :submission} &&
+          incomes.all? {|income| income.nil? || income.amount > 0 } &&
```

Comment:
`income.amount > 0 `works for nil incomes, so we may not need the` income.nil?` check.

Reply:
No we would, there's no `>` method for `nil` so good job here.